buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -128,7 +136,18 @@ public void apply(Project project) {
                     TaskProvider<GradleDistroTestTask> vmTask =
                         configureVMWrapperTask(vmProject, distribution.getName() + " distribution", destructiveTaskName, vmDependencies);
                     vmTask.configure(t -> t.dependsOn(distribution));
-                    distroTest.configure(t -> t.dependsOn(vmTask));
+
+                    distroTest.configure(t -> {
+                        // Only VM sub-projects that are specifically opted-in to testing Docker should
+                        // have the Docker task added as a dependency. The shouldTestDocker property could
+                        // be null, hence we use Boolean.TRUE.equals()
+                        boolean shouldExecute = distribution.getType() != Type.DOCKER
```

Comment:
The typical way to go about this is to register an Gradle extension on the project where the project could provide additional configuration, like weather or not to test on docker. 

Aslo, I think we should specifically opt-put  instead of opt in as most platforms support it and future platforms are expected to support it.

Reply:
Not sure what earlier comment you are referring to @atorok. 

Also, I think this is going to have to go in an `afterevaluate { }` block to ensure that property has been evaluated.

Reply:
I was referring to this one: https://github.com/elastic/elasticsearch/pull/47640#discussion_r340652049 didn't realize GH will put it after not before. 

My point is that we don't need to bother with docker support for the subprojects for when docker is not available in the Vagrant VM ( which is when these are actually used ) . The tests are skipped when there's no docker and there is no need to build an image inside the VM, so we're all good. Not having docker on the host that runs vagrant is ok to error out.